File: src/main/java/mnm/mods/tabbychat/client/settings/GeneralSettings.java
package mnm.mods.tabbychat.client.settings;

import mnm.mods.tabbychat.util.TimeStamps;
import mnm.mods.tabbychat.util.config.Value;
import mnm.mods.tabbychat.util.config.ValueObject;
import net.minecraft.util.text.TextFormatting;

public class GeneralSettings extends ValueObject {

    public Value<Boolean> logChat = value(true);
    public Value<Boolean> splitLog = value(true);
    public Value<Boolean> timestampChat = value(false);
    public Value<TimeStamps> timestampStyle = value(TimeStamps.MILITARYSECONDS);
    public Value<TextFormatting> timestampColor = value(TextFormatting.WHITE);
    public Value<Boolean> antiSpam = value(false);
    public Value<Double> antiSpamPrejudice = value(0D);
    public Value<Boolean> unreadFlashing = value(true);
    public Value<Boolean> checkUpdates = value(true);
}

File: src/main/kotlin/mnm/mods/tabbychat/client/settings/GeneralSettings.kt
package mnm.mods.tabbychat.client.settings

import mnm.mods.tabbychat.util.TimeStamps
import mnm.mods.tabbychat.util.config.AbstractConfigView
import mnm.mods.tabbychat.util.config.Comment
import mnm.mods.tabbychat.util.config.ConfigView
import net.minecraft.util.text.TextFormatting

class GeneralSettings(config: AbstractConfigView, path: List<String>) : ConfigView(config, path) {
    @Comment("Prepend chat messages with the timestamp")
    var timestampChat by defining(false)

    @Comment("The style of the timestamp")
    var timestampStyle by definingEnum(TimeStamps.MILITARYSECONDS, TimeStamps::class.java)

    @Comment("The color of the timestamp")
    var timestampColor by definingRestrictedEnum(TextFormatting.WHITE, TextFormatting::class.java, TextFormatting.values().filter { it.isColor })

    @Comment("Show flashing notifications when a channel has unread messages")
    var unreadFlashing by defining(true)

    @Comment("Enable update checking (not implemented)")
    var checkUpdates by defining(true)
}
